tests/Functional/Schema/PostgreSQLSchemaManagerTest.php

```diff
@@ -561,6 +564,73 @@ public function testAlterTableAutoIncrementIntToBigInt(
         self::assertTrue($tableFinal->getColumn('id')->getAutoincrement());
     }
 
+    public function testListTableColumnsOidConflictWithNonTableObject(): void
+    {
+        $versionParts      = [];
+        $wrappedConnection = $this->connection->getWrappedConnection();
+        assert($wrappedConnection instanceof ServerInfoAwareConnection);
+        assert(preg_match('/^(?P<major>\d+)/', $wrappedConnection->getServerVersion(), $versionParts) > 0);
```

Comment:
Since we're going to use `$versionParts` afterwards, we need to be sure the `preg_match()` is actually executed. Since `assert()`s can be disabled, you'll have to move that call out of the `assert()`.

That being said, you might just use PHP's `version_compare()` function to check if we're on Postgres 10 or later.

Reply:
I'm not sure if I implemented what you wanted.  If `assert` can be disabled, should I throw an exception if `$versionParts['major']` isn't set?  That's what I was trying to accomplish with the original `assert` call.